fabric-item-api-v1/src/main/java/net/fabricmc/fabric/api/item/v1/FabricItem.java

```diff
@@ -90,4 +90,18 @@ default Multimap<EntityAttribute, EntityAttributeModifier> getAttributeModifiers
 	default boolean isSuitableFor(ItemStack stack, BlockState state) {
 		return ((Item) this).isSuitableFor(state);
 	}
+
+	/**
+	 * Returns a leftover item stack after {@code stack} is consumed in a recipe.
+	 * (This is also known as "recipe remainder".)
+	 * For example, using a lava bucket in a furnace as fuel will leave an empty bucket.
+	 *
+	 * <p>This is a stack-aware version of {@link Item#getRecipeRemainder()}.
+	 *
+	 * @param stack the consumed {@link ItemStack}
+	 * @return the leftover item stack
+	 */
```

Comment:
```suggestion
	/**
	 * Returns a leftover item stack after {@code stack} is consumed in a recipe.
	 * (This is also known as "recipe remainder".)
	 * For example, using a lava bucket in a furnace as fuel will leave an empty bucket.
	 *
	 * <p>The passed {@code stack} should not be modified, and its count should be ignored.
	 * The returned stack, if not empty, must be a new instance with count 1.
	 *
	 * <p>This is a stack-aware version of {@link Item#getRecipeRemainder()}.
	 * Using {@link FabricItemStack#hasRecipeRemainder} and {@link FabricItemStack#getRecipeRemainder} instead of calling this function directly is recommended.
	 *
	 * @param stack the consumed {@link ItemStack}
	 * @return the leftover item stack
	 */
```